app/src/main/java/com/github/orelzion/spotifyclone/view/AlbumListFragment.kt

```diff
@@ -3,32 +3,63 @@ package com.github.orelzion.spotifyclone.view
 import android.os.Bundle
 import android.util.Log
 import android.view.View
+import android.widget.ProgressBar
 import android.widget.Toast
+import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
-import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.fragment.app.commit
 import androidx.recyclerview.widget.RecyclerView
 import com.github.orelzion.spotifyclone.R
 import com.github.orelzion.spotifyclone.model.AlbumsResponse
 import com.github.orelzion.spotifyclone.model.repository.BrowseRepository
 import com.github.orelzion.spotifyclone.viewmodel.AlbumViewData
 
+
 /**
  * A fragment representing a list of Items.
  */
-class AlbumListFragment : Fragment(R.layout.fragment_album_list) {
+class AlbumListFragment : Fragment(R.layout.fragment_items_list) {
 
-    private val albumsListView: RecyclerView by lazy { requireView().findViewById(R.id.albumsList) }
+    private val albumsListView: RecyclerView by lazy { requireView().findViewById(R.id.itemsList) }
     private val albumsAdapter: AlbumsAdapter by lazy { AlbumsAdapter() }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        albumsListView.adapter = albumsAdapter
-        albumsListView.layoutManager = LinearLayoutManager(requireContext())
+        /**
+         * Set albumAdapter's clickListener to AlbumListFragment's onItemClicked method
+         */
+        albumsListView.adapter = albumsAdapter.apply {
+            clickListener = this@AlbumListFragment::onItemClicked
+        }
 
+        /**
+         * (re-)load Albums - depends on it's availability
+         */
         if (savedInstanceState == null) {
             loadAlbums()
         }
+        // TODO: add else - restore savedInstanceState rather than (re-)loadAlbums
+
+        requireView().findViewById<ProgressBar>(R.id.progressBar).isVisible = false
+    }
+
+    /**
+     * onItemClicked - show corresponding tracks' list in main fragment
+     */
+    private fun onItemClicked(albumId: String) {
+        showFragment(R.id.mainFragmentContainer, TracksListFragment.newInstance(albumId), true)
+    }
+
+    // TODO: Ask Orel - is this method should be declared independently,
+    //  so it stays available for both AlbumListFragment and TracksListFragment?
```

Comment:
So theoretically yes, we could have written a shared function here, but we would need to replace it outside of this fragment in order for that to be shared.
One way will be to have a `BaseFragment` class with this function and have them both inherit from it, but I find it too limiting.
Another option will be to use extension functions, which is something we didn't learn yet

Reply:
Nice :)
I'll wait for the extentions..